apps/avatax/src/modules/avatax/calculate-taxes/avatax-calculate-taxes-payload-lines-transformer.ts

```diff
@@ -16,16 +81,26 @@ export class AvataxCalculateTaxesPayloadLinesTransformer {
      * https://developer.avalara.com/erp-integration-guide/sales-tax-badge/transactions/discounts-and-overrides/discounting-a-transaction/
      */
     const isDiscounted = taxBase.discounts.length > 0;
+    const { hasEntireCheckoutDiscount, hasOncePerOrderVoucher } = checkDiscounts(
+      taxBase,
+      isDiscounted,
+    );
+
     const productLines: LineItemModel[] = taxBase.lines.map((line) => {
       const matcher = new AvataxCalculateTaxesTaxCodeMatcher();
       const taxCode = matcher.match(line, matches);
 
       return {
-        amount: line.totalPrice.amount,
+        amount: getUndiscountedTotalPrice(line),
         taxIncluded: taxBase.pricesEnteredWithTax,
         taxCode,
         quantity: line.quantity,
-        discounted: isDiscounted,
+        discounted: isProductLineDiscounted(
+          line,
+          isDiscounted,
+          hasOncePerOrderVoucher,
+          hasEntireCheckoutDiscount,
+        ),
```

Comment:
please add suite of unit tests to check these changes. there are a lot of "ifs" hidden out there

e2e tests can follow some basic scenarios but on unit tests level we can check variety of edge cases